compiler/main/driver.cpp

```diff
@@ -1367,6 +1368,35 @@ static void postVectorize() {
   }
 }
 
+static void setMultiLocaleInterop() {
+  // We must be compiling a multi-locale library to be eligible for MLI.
+  if (!fLibraryCompile || !strcmp(CHPL_COMM, "none")) {
+    return;
+  }
+
+  if (strcmp(CHPL_COMM, "gasnet") != 0) {
+    USR_FATAL("Multi-locale libraries are only supported on gasnet");
+    return;
+  }
+
+  if (llvmCodegen) {
+    USR_FATAL("Multi-locale libraries do not support --llvm");
+    return;
+  }
+
+  if (fLibraryPython) {
+    USR_FATAL("Multi-locale libraries do not support --library-python");
+    return;
+  }
+
+  if (fLibraryFortran) {
+    USR_FATAL("Multi-locale libraries do not support --library-fortran");
+    return;
```

Comment:
Note that the return statements aren't necessary here as USR_FATAL will cause the compiler to end immediately.  But it's fine to leave them

Reply:
Good catch!